### PR TITLE
Handle loading state

### PR DESCRIPTION
**Pull request by ionnikovv**
In this branch was added handling of loading state

### CODE REVIEW

**Review by TheForsakenSpirit — APPROVED** [review]

File: src/components/PokemonCard/PokemonCard.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ export function PokemonCard({ pokemonItem }: Props): JSX.Element {
   const [pokemonInfo, setPokemonInfo] = useState({ weight: 0, baseExperience: 0, height: 0 });
   const [isLoading, setIsLoading] = useState(true);
```
Comment (mrlika) [inline src/components/PokemonCard/PokemonCard.tsx:13]: Set `isLoading` to `true` when loading started

File: src/components/PokemonCard/PokemonCard.tsx [hunk 2]
```diff
@@ -30,11 +31,14 @@ export function PokemonCard({ pokemonItem }: Props): JSX.Element {
...
   }, [pokemonItem.url]);
 
-  return (
-    <div className='pokemon-card' key={pokemonItem.name}>
-      {isLoading ? (
-        <Loader />
-      ) : (
+  if (isLoading) return <Loader />;
+  else
+    return (
```
Comment (mrlika) [inline src/components/PokemonCard/PokemonCard.tsx:36]: Add curly brackets

File: src/App.tsx [hunk 3]
```diff
@@ -1,50 +1,54 @@
...
-
+import { useEffect, useState } from 'react';
+import { Loader } from './components/Common/Loader/Loader';
 import { PokemonCard } from './components/PokemonCard/PokemonCard';
 import { useIntersectionObserver } from './Hooks/useOnScreen';
 import { isPokemonResponse, PokemonsData } from './types/PokemonsData';
+import './App.css';
 
-const PAGE_SIZE = 10;
+const PAGE_SIZE = 1;
```
Comment (mrlika) [inline src/App.tsx:8]: Leave `PAGE_SIZE = 10`

File: src/App.tsx [hunk 4]
```diff
@@ -1,50 +1,54 @@
...
       <div className='pokemon-cards'>
         {pokemons.map((pokemonItem) => (
           <PokemonCard pokemonItem={pokemonItem} key={pokemonItem.name} />
         ))}
-        <div ref={refObserver} className='observer-block'></div>
+        {isFetching && <Loader />}
+        <div ref={setRef} className='observer-block' />
       </div>
     </div>
   );
```
Comment (mrlika) [inline src/App.tsx:54]: Don't export default: https://basarat.gitbook.io/typescript/main-1/defaultisbad

File: src/App.tsx [hunk 5]
```diff
@@ -1,50 +1,54 @@
...
 import { isPokemonResponse, PokemonsData } from './types/PokemonsData';
+import './App.css';
 
-const PAGE_SIZE = 10;
+const PAGE_SIZE = 1;
 
 function App(): JSX.Element {
   const [pokemons, setPokemons] = useState<PokemonsData[]>([]);
-  const [lastPage, setLastPage] = useState(0);
+  const [lastPage, setLastPage] = useState(1);
```
Comment (mrlika) [inline src/App.tsx:12]: Rename to `currentPage`

File: src/App.tsx [hunk 6]
```diff
@@ -1,50 +1,54 @@
...
   const [isFetching, setIsFetching] = useState(true);
-  const refObserver = useRef<HTMLDivElement>(null);
+  const [ref, setRef] = useState<HTMLDivElement | null>(null);
 
   useEffect(() => {
+    if (lastPage === 0) return;
```
Comment (mrlika) [inline src/App.tsx:17]: Remove this redundant check

File: src/App.tsx (hunk 5 above)
Comment (mrlika) [inline src/App.tsx:12]: Leave counting pages from 0

File: src/App.tsx [hunk 7]
```diff
@@ -1,50 +1,54 @@
...
     async function getPokemons() {
       const response = await fetch(
-        `https://pokeapi.co/api/v2/pokemon?limit=${PAGE_SIZE}&offset=${lastPage * PAGE_SIZE}`
+        `https://pokeapi.co/api/v2/pokemon?limit=${PAGE_SIZE}&offset=${(lastPage - 1) * PAGE_SIZE}`
```
Comment (mrlika) [inline src/App.tsx:20]: Don't change the line if counting from 0

File: src/App.tsx [hunk 8]
```diff
@@ -1,50 +1,54 @@
...
+
   return (
     <div className='pokemon-wrapper'>
       <span className='pokemon-logo'>choose your pokemon!</span>
       <div className='pokemon-cards'>
         {pokemons.map((pokemonItem) => (
           <PokemonCard pokemonItem={pokemonItem} key={pokemonItem.name} />
         ))}
-        <div ref={refObserver} className='observer-block'></div>
+        {isFetching && <Loader />}
```
Comment (mrlika) [inline src/App.tsx:50]: Remove this line

File: package-lock.json [hunk 9]
```diff
@@ -5223,6 +5223,11 @@
         }
       }
     },
+    "email-addresses": {
```
Comment (mrlika) [inline package-lock.json:5226]: Remove this file

**Review by mrlika — CHANGES_REQUESTED** [review]

**Review by mrlika — APPROVED** [review]

**Pull request merged**